packages/dds/tree/src/util/utils.ts

```diff
@@ -12,6 +12,19 @@ export type RecursiveReadonly<T> = {
     readonly [P in keyof T]: RecursiveReadonly<T[P]>;
 };
 
+/**
+ * Convert a union type to an intersection type
+ * @example
+ * ```typescript
+ * type T = UnionToIntersection<number | string> // `number & string`
+ * ```
+ */
+export type UnionToIntersection<T> = (T extends any ? (k: T) => void : never) extends (
```

Comment:
I suspect anything that depends on this isn't going to be very sound.

For example if we have
```typescript
type X = A | B;
type Y = C | D;
type Z = UnionToIntersection<X | Y>; // Should this be X & Y? A & B & C & D?
```

We can also consider other cases, like:
```typescript
type Number1 = never | number; // This type is identical to just number
type Number2 = number | number; // This type is also identical to just number
UnionToIntersection<Number1>; // This should give `never`?
UnionToIntersection<Number1>; // This should give `number`?
```
Is `UnionToIntersection<bool>;` supposed to be `true & false` aka `never`?

Is `UnionToIntersection<bool | bool>;` supposed to be `bool`?

In short: according to its documentation, it sounds like this is supposed to produce different results when given multiple equivalent types. Anything that ties to do that is going to be flaky because type normalization impacts how it type checks.

If you think of a type as defining a set of allowed values (ex the type number is a union of all possible numeric values) this generic type makes no sense.

Another different case:
```typescript
interface Foo {
bar: number | string;
baz?: number;
}
// depending on context, this can get interpreted/normalized many ways, for example in a function which tests baz, it maps to:
{
bar: number | string;
baz: undefined;
} | {
bar: number | string;
baz: bool;
}
// but if you type test bar, you get
{
bar: string;
baz?: bool;
} | {
bar: number ;
baz?: bool;
}

UnionToIntersection<Foo> // SO which way should UnionToIntersection work?
```

Reply:
Great points Craig. I think I added this to our utils reflexively simply because it can be factored out, but that's not a good reason; you're making me realize it should probably just remain private to the module that uses it for the event transformations. It happens to be useful for the event transforms, but it's not a very good "general purpose" tool. I'll hide it.